interface/resources/qml/hifi/avatarapp/AvatarsModel.qml

```diff
@@ -21,17 +21,47 @@ ListModel {
 
         return marketItemUrl;
     }
+	
+    function makeMarketThumbnailUrl(marketId) {
+        var avatarThumbnailUrl = "https://hifi-metaverse.s3-us-west-1.amazonaws.com/marketplace/previews/%marketId%/large/hifi-mp-%marketId%.jpg"
+            .split('%marketId%').join(marketId);
+            
+        return avatarThumbnailUrl;
+    }
+	
+    function trimFileExtension(url) {
+        var trimmedUrl = url.substring(0, (url.indexOf("#") == -1) ? url.length : url.indexOf("#"));
+        trimmedUrl = trimmedUrl.substring(0, (trimmedUrl.indexOf("?") == -1) ? trimmedUrl.length : trimmedUrl.indexOf("?"));
+        trimmedUrl = trimmedUrl.substring(0, trimmedUrl.lastIndexOf("."));
+
+        return trimmedUrl;
+    }
+	
+    function imageExists(image_url){
+
+        var http = new XMLHttpRequest();
+
+        http.open('HEAD', image_url, false);
```

Comment:
`HEAD` is not a valid HTTP method, I suspect you meant to put `GET`

Reply:
No, HEAD is perfectly legitimate. It's a GET without content, only headers. See for instance:

https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

Reply:
Seems to save bandwidth, thanks for clarifying good to know this!